src/box/applier.cc

```diff
@@ -863,8 +863,14 @@ applier_fetch_snapshot(struct applier *applier)
 	struct iostream *io = &applier->io;
 	struct xrow_header row;
 
+	struct vclock vclock;
+	vclock_create(&vclock);
 	struct fetch_snapshot_request req = {
 		.version_id = tarantool_version_id(),
+		/* Applier doesn't support checkpoint join. */
+		.is_checkpoint_join = false,
+		.checkpoint_vclock = vclock,
+		.checkpoint_lsn = 0,
```

Comment:
Why do you specify empty vclock then? Is this field mandatory? Otherwise you can drop these initializations from here. The struct initialization in the form of `var = { .field1 = ..., .field2 = ... ...}` will nullify all non-specified fields.

Reply:
> The struct initialization in the form of var = { .field1 = ..., .field2 = ... ...} will nullify all non-specified fields.

Our compilation flags forbid doing this:

```
/home/serpentian/Programming/tnt/tarantool-ee/tarantool/src/box/applier.cc: In function ‘void applier_fetch_snapshot(applier*)’:
/home/serpentian/Programming/tnt/tarantool-ee/tarantool/src/box/applier.cc:874:9: error: missing initializer for member ‘fetch_snapshot_request::checkpoint_vclock’ [-Werror=missing-field-initializers]
  874 |         };
      |         ^
```

Moreover, you and Vladimir were against unitialized fields in https://github.com/tarantool/tarantool/pull/9973#discussion_r1605324874.

I dropped `vclock_create` since it really is not mandatory, but the structure must be fully initialized, if we don't wanna use `req.checkpoint_lsn = ...`